File: Buddha-Hr-frontend/buddha-hr-frontend/src/services/companyService.js
import api from "./api";

export const getCompanies = async (
  page = 0,
  size = 10,
  sortBy = "name",
  direction = "asc"
) => {
  const response = await api.get("/companies", {
    params: {
      page,
      size,
      sortBy,
      direction,
    },
  });

  return response.data;
};

export const getCompany = async (id) => {
  const response = await api.get(`/companies/${id}`);
  return response.data;
};

export const createCompany = async (formData) => {
  const response = await api.post("/companies", formData);

  return response.data;
};

export const updateCompany = async (id, formData) => {
  const response = await api.put(`/companies/${id}`, formData, {
    headers: {
      "Content-Type": "multipart/form-data",
    },
  });

  return response.data;
};

export const deleteCompany = async (id) => {

  const response = await api.delete(`/companies/${id}`);

  return response.data;

};

export const searchCompanies = async (keyword) => {
  const response = await api.get("/companies/search", {
    params: { keyword },
  });

  return response.data;
};

File: buddha-hr-backend/buddha-hr-backend/src/main/java/com/buddhahr/company/service/CompanyService.java
package com.buddhahr.company.service;

import com.buddhahr.company.dto.request.CompanyRequest;
import com.buddhahr.company.dto.response.CompanyResponse;
import org.springframework.data.domain.Page;
import org.springframework.web.multipart.MultipartFile;

import java.io.IOException;

public interface CompanyService {

    CompanyResponse createCompany(
            CompanyRequest request,
            MultipartFile logo
    ) throws IOException;

    CompanyResponse updateCompany(Long id,
                  CompanyRequest request,
                  MultipartFile logo)throws IOException;

    void deleteCompany(Long id);

    CompanyResponse getCompany(Long id);

    Page<CompanyResponse> getAllCompanies(
            int page,
            int size,
            String sortBy,
            String direction
    );

    Page<CompanyResponse> searchCompanies(
            String keyword,
            int page,
            int size
    );

    Page<CompanyResponse> getActiveCompanies(
            int page,
            int size
    );

}
